lib/portlayer/exec2/remote/server/portlayer_rpc_server.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"encoding/hex"
+	"fmt"
+	"log"
+	"math/rand"
+	"net"
+	"net/http"
+	"net/rpc"
+
+	"github.com/vmware/vic/lib/portlayer/exec2"
+	"github.com/vmware/vic/lib/portlayer/exec2/remote"
+)
+
+type PortLayerRPCServer struct {
+}
+
+// A Handle can be anything, so this takes advantage of that by creating a sparse handle
+// to send to the client and using that sparse handle as a key in a hashtable which points to
+// rich handles created by the HandleFactory
+type SparseHandle exec2.Handle
+
+var lcTarget exec2.ContainerLifecycle
+
+//var lcQuery exec2.ContainerQuery
+var handles map[SparseHandle]exec2.Handle
```

Comment:
So I asked this question since this kind of resource management may affect how the handle is represented and passed back to the client. Should we store something on the server side vs. a stateless design where the client holds that information.